File: Backend/Tafe/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tafe.Models
{
    public class Order : EntityTemplate
    {
        [Required]
        public string OrderNumber { get; set; }
            = Guid.NewGuid().ToString("N")[..8].ToUpper();

        // Customer is optional => Guest Order
        public string? CustomerId { get; set; }

        public virtual CustomerProfile? Customer { get; set; }

        // The employee who created the order
        [Required]
        public required string CashierId { get; set; }

        public virtual required ApplicationUser Cashier { get; set; }

        // Optional for TakeAway / Delivery
        public int? TableId { get; set; }

        public virtual CafeTable? Table { get; set; }

        public OrderType OrderType { get; set; }

        public OrderStatus Status { get; set; }
            = OrderStatus.Pending;

        [Column(TypeName = "decimal(18,2)")]
        public decimal SubTotal { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Discount { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Tax { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Service { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Total { get; set; }

        public virtual ICollection<OrderItem> Items { get; set; } = [];

        public virtual ICollection<Payment> Payments { get; set; } = [];
    }
}

File: Frontend/src/types/order.ts
export type OrderType =
  | "DineIn"
  | "TakeAway"
  | "Delivery";

export type OrderStatus =
  | "Pending"
  | "Preparing"
  | "Ready"
  | "Completed"
  | "Cancelled";

export interface OrderItem {
  id: number;
  orderId: number;
  productId: number;
  productName?: string;

  quantity: number;
  unitPrice: number;
  discount: number;
  total: number;
  notes?: string;
}

export interface Order {
  id: number;
  orderNumber: string;

  customerId?: string | null;
  customerName?: string | null;

  cashierId?: string | null;
  cashierName?: string | null;

  tableId?: number | null;
  tableName?: string | null;

  orderType: OrderType;
  status: OrderStatus;

  subTotal: number;
  discount: number;
  tax: number;
  service: number;
  total: number;

  createdAt: string;
  updatedAt: string;

  items: OrderItem[];
}

export interface OrderItemCreate {
  productId: number;
  quantity: number;
  discount: number;
  notes?: string;
}

export interface OrderCreate {
  customerId?: string | null;
  tableId?: number | null;
  orderType: OrderType;

  discount: number;
  tax: number;
  service: number;

  items: OrderItemCreate[];
}

export interface OrderUpdate {
  id: number;

  customerId?: string | null;
  orderType?: OrderType;
  tableId?: number;

  discount?: number;
  tax?: number;
  service?: number;
}

export interface OrderStatusUpdate {
  id: number;
  status: OrderStatus;
}

export interface OrderItemUpdate {
  id: number;
  productId: number;
  quantity: number;
  discount: number;
  notes?: string;
}